feat(js): Document `onlyIncludeInstrumentedModules` option (#11632)
getsentry/sentry-docs: docs/platforms/javascript/common/install/esm.mdx

We do not support ESM in Node versions before 18.19.0.

## Troubleshooting instrumentation

By default, all packages are wrapped under the hood by
[import-in-the-middle](https://www.npmjs.com/package/import-in-the-middle) to
aid instrumenting them.

If `import-in-the-middle` encounters problems wrapping a package, you may see
syntax errors at runtime or logged errors in your console:

```logs
SyntaxError: The requested module '...' does not provide an export named '...'
(node:3368) Error: 'import-in-the-middle' failed to wrap 'file://../../path/to/file.js'
```

> You can confirm that these errors are caused by `import-in-the-middle` by

Reviewer: ```suggestion
To confirm that these errors are caused by `import-in-the-middle`,
```

> disabling it by setting `registerEsmLoaderHooks` to false. Note, this will also

Reviewer: ```suggestion
disable it by setting `registerEsmLoaderHooks` to false. Note, this will also
```
